chainer/utils/sparse.py

```diff
@@ -1,5 +1,14 @@
+import numpy
+
 import chainer
 from chainer import backend
+from chainer import cuda
+
+
+def _add_at(add_at, x, row, col, data):
+    assert data.size > 0
+    last_nz = data.size - (data != 0)[::-1].argmax()
```

Comment:
This should correctly handle intermediate 0s in the data array, in contrast to the previous `nnz = xp.count_nonzero(data[i])`.